product_configurator/models/product.py

```diff
@@ -369,19 +370,26 @@ def value_available(self, attr_val_id, value_ids):
             lambda l: attr_val_id in l.value_ids.ids
         )
 
-        domains = config_lines.mapped('domain_id').compute_domain()
-
-        for domain in domains:
-            if domain[1] == 'in':
-                if not set(domain[2]) & set(value_ids):
-                    return False
-            else:
-                if set(domain[2]) & set(value_ids):
-                    return False
+        for config_line in config_lines:
+            domains = config_line.mapped('domain_id').compute_domain()
+            for domain in domains:
+                if domain[1] == 'in':
+                    if not set(domain[2]) & set(value_ids):
+                        # Don't break if the module has not been upgraded...
+                        if do_raise and config_line.rule_description:
```

Comment:
Ahh, not, not quite.  What I discovered is, that it is in fact redundant!  It fails in any case. Sure, upgrading the module puts the fields in the db, but restarting the server finds the field definitions and therefore puts them in the _fields list.  So in fact, running the code without upgrading the module still has an issue...